coffee/base.py

```diff
@@ -435,24 +435,27 @@ class Symbol(Expr):
     * 1: array
     * 2: matrix, etc.
 
-    :param tuple rank: entries represent the iteration variables the symbol
+    :param symbol: the Symbol name.
+    :param rank: entries represent the iteration variables the symbol
         depends on, or explicit numbers representing the entry of a tensor the
         symbol is accessing, or the size of the tensor itself.
-    :param list offset: an iterator of 2-tuple (period, stride), one 2-tuple for
+    :param offset: an iterator of 2-tuple (period, stride), one 2-tuple for
         each entry in rank. The period is the multiplier of the rank, while
         stride is a quantity summed to the rank. E.g., if rank=(i, j) and
         offset=((1, 0), (3, 2)), printing a symbol 'a' returns the string
         a[i][3*j + 2].
+    :param operator: an optional C operator to be applied (e.g., *, &).
     """
 
-    def __init__(self, symbol, rank=(), offset=()):
+    def __init__(self, symbol, rank=(), offset=(), operator=None):
         super(Symbol, self).__init__([])
         self.symbol = symbol
         self.rank = rank
         self.offset = offset or tuple([(1, 0) for r in rank])
+        self.operator = operator
 
     def operands(self):
-        return [self.symbol, self.rank, self.offset], {}
+        return [self.symbol, self.rank, self.offset, self.operator], {}
```

Comment:
I think that that ad-hoc inheritance would be even more convoluted than the heavyweight `Symbol` class we currently have.

The right thing to do, of course, is not to be afraid of refactoring.

> there are places in which a Symbol is expected, and now an Operator could actually pop up

Clearly, they can handle more just a symbol if they work with the heavyweight "symbols" that we now have. What are they really expecting? Maybe you can introduce an abstract class for that concept.

Or is it just about skipping certain "symbol modifiers" when checking a subexpression? Then you could just write some utility functions, e.g. `is_modified_symbol`, instead of sticking to `isinstance(..., Symbol)` and lazily adding the kitchen sink to the `Symbol` class.